plugins/notion/src/blocksToHTML.ts

```diff
@@ -94,6 +94,24 @@ export function blocksToHtml(blocks: BlockObjectResponse[]) {
             case "code":
                 htmlContent += `<pre><code class="language-${block.code.language.replace(" ", "-")}">${richTextToHTML(block.code.rich_text)}</code></pre>`
                 break
+            case "video":
+                if (block.video.type !== "external") {
+                    break
+                }
+
+                const videoUrl = block.video.external.url
+                const youtubeMatch = videoUrl.match(
+                    /(?:youtube\.com\/(?:[^\/]+\/.+\/|(?:v|e(?:mbed)?)\/|.*[?&]v=)|youtu\.be\/)([^"&?\/\s]{11})/i
+                )
+
+                const videoId = youtubeMatch?.[1]
+                if (!videoId) {
+                    break
+                }
+
+                // Framer styles and modifies the YouTube iframe automatically
+                htmlContent += `<iframe src="https://www.youtube.com/embed/${videoId}"></iframe>`
+                break
```

Comment:
Would be easier to read like this, wdyt?

```suggestion
                const youtubeId = videoUrl.match(YOUTUBE_ID_REGEX)?.groups?.videoId
                if (youtubeId) {
                    htmlContent += `<iframe src="https://www.youtube.com/embed/${youtubeId}"></iframe>`
                }
                break
```